### PR TITLE
Add image to Prop Card

### PR DESCRIPTION
**Pull request by cdt-eth**
If a proposal has an image we'll display it on the prop card itself. Note, if the image url is invalid it won't show the image.

Screenshot 2023-01-25 at 10 44 09 AM

### CODE REVIEW

File: packages/prop-house-webapp/src/utils/getImageFromDescription.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { StoredProposalWithVotes } from '@nouns/prop-house-wrapper/dist/builders';
+import isImage from 'is-image-url';
+
+const getImageFromDescription = async (proposal: StoredProposalWithVotes) => {
+  // proposal description field
+  const what = proposal.what;
```
Comment (cryptoseneca) [inline packages/prop-house-webapp/src/utils/getImageFromDescription.ts:6]: good idea to use "proposalDescription" here lol

File: packages/prop-house-webapp/src/utils/getImageFromDescription.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  // check if there's an image in the description
+  if (match) {
+    try {
+      // use is-image-url to check if the image is valid
+      const isValid = await isImage(match[1]);
+      if (isValid) {
+        // return the image url
+        return match[1];
+      } else {
+        return '';
```
Comment (cryptoseneca) [inline packages/prop-house-webapp/src/utils/getImageFromDescription.ts:23]: should we return `undefined` instead? seems like that would make more sense.

File: packages/prop-house-webapp/src/components/ProposalCard/index.tsx [hunk 3]
```diff
@@ -102,31 +131,38 @@ const ProposalCard: React.FC<{
...
-                    <span className={classes.plusArrow}>+</span>
-                    <PropCardVotingModule proposal={proposal} />
+            {(auctionStatus === AuctionStatus.AuctionVoting ||
+              auctionStatus === AuctionStatus.AuctionEnded) && (
+              <div className={classes.timestampAndlinkContainer}>
+                <div
+                  className={clsx(
+                    classes.avatarAndPropNumber,
+                    !roundIsVotingOrOver() && classes.hideVoteModule,
+                    // classes.hideVoteModule,
```
Comment (cryptoseneca) [inline packages/prop-house-webapp/src/components/ProposalCard/index.tsx:151]: remove comment

File: packages/prop-house-webapp/src/utils/getImageFromDescription.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+  // find the first image in the description
+  const match = imgRegex.exec(what);
+
+  // check if there's an image in the description
+  if (match) {
+    try {
+      // use is-image-url to check if the image is valid
+      const isValid = await isImage(match[1]);
+      if (isValid) {
```
Comment (cryptoseneca) [inline packages/prop-house-webapp/src/utils/getImageFromDescription.ts:19]: nit: `return isValid ? match[1] : undefined`

**Review by cryptoseneca — COMMENTED** [review]
One item regarding UX: let's make ENS/address have the same font appearance as proposal creation date (i think date is bold while ens isn't)
Screen Shot 2023-01-25 at 11 06 59 AM

File: packages/prop-house-webapp/src/utils/getImageFromDescription.ts (hunk 1 above)
Comment (cdt-eth) [reply packages/prop-house-webapp/src/utils/getImageFromDescription.ts]: https://github.com/Prop-House/prop-house-monorepo/pull/407/commits/2b59d49574c42f1f0d7d2081e81753ebcee38cb8

File: packages/prop-house-webapp/src/utils/getImageFromDescription.ts (hunk 4 above)
Comment (cdt-eth) [reply packages/prop-house-webapp/src/utils/getImageFromDescription.ts]: https://github.com/Prop-House/prop-house-monorepo/pull/407/commits/b762d1136c8706efcf9ad127529c5370a01d1702

File: packages/prop-house-webapp/src/components/ProposalCard/index.tsx (hunk 3 above)
Comment (cdt-eth) [reply packages/prop-house-webapp/src/components/ProposalCard/index.tsx]: https://github.com/Prop-House/prop-house-monorepo/pull/407/commits/d5952f1a00630a1a4beefec3eec3bef4b941b18b

**Comment by cdt-eth** [comment]
> One item regarding UX: let's make ENS/address have the same font appearance as proposal creation date (i think date is bold while ens isn't) Screen Shot 2023-01-25 at 11 06 59 AM
 

> Screen Shot 2023-01-25 at 11 09 30 AM

https://github.com/Prop-House/prop-house-monorepo/pull/407/commits/aa616c40b4b8df9f92c3fa8969c6f07bc69a061d

**Pull request merged**